ViewModel/HyvaGridViewModel.php

```diff
@@ -187,12 +190,15 @@ private function createCellWithRowExcludingCell(HyvaGrid\CellInterface $cell, ar
 
     public function getNavigation(): HyvaGrid\NavigationInterface
     {
-        return $this->navigationFactory->create([
-            'gridName'          => $this->getGridName(),
-            'gridSource'        => $this->getGridSourceModel(),
-            'columnDefinitions' => $this->getColumnDefinitions(),
-            'navigationConfig'  => $this->getGridDefinition()->getNavigationConfig(),
-        ]);
+        if (!isset($this->memorizedNavigation)) {
+            $this->memorizedNavigation = $this->navigationFactory->create([
+                'gridName'          => $this->getGridName(),
+                'gridSource'        => $this->getGridSourceModel(),
+                'columnDefinitions' => $this->getColumnDefinitions(),
+                'navigationConfig'  => $this->getGridDefinition()->getNavigationConfig(),
+            ]);
+        }
+        return $this->memorizedNavigation;
```

Comment:
Did you profile if the navigation memoization adds any measurable benefits?

Reply:
Yes, i did because i would like to use the navigation like $grid->getNavigation() and if you use it a second time (which i did for the exports) the whole factory is creating everything again which is quite consuming